serve.go

```diff
@@ -9,25 +9,37 @@ import (
 	"time"
 )
 
-// Run starts the server.
-// It is blocking.
-// It returns an error if the server could not start (it could not bind to the port for example).
-// It also generates the OpenAPI spec and outputs it to a file, the UI, and a handler (if enabled).
-func (s *Server) Run() error {
+func (s *Server) setupRun(proto string) {
 	go s.OutputOpenAPISpec()
 	elapsed := time.Since(s.startTime)
 	slog.Debug("Server started in "+elapsed.String(), "info", "time between since server creation (fuego.NewServer) and server startup (fuego.Run). Depending on your implementation, there might be things that do not depend on fuego slowing start time")
-	slog.Info("Server running ✅ on http://"+s.Server.Addr, "started in", elapsed.String())
+	slog.Info("Server running ✅ on "+proto+"://"+s.Server.Addr, "started in", elapsed.String())
```

Comment:
Most of the time I just bind my servers to all interfaces using the shorthand syntax `:8080` so this log message becomes 
> Server running ✅ on http://:8080 
witch bugs me a bit. 
I also noticed Fuego deprecated the `WithPort` option in favor of the `WithAddress` one, so maybe we want to  update this log message to better reflect that?

Reply:
What do you have in mind?

Reply:
Also when it comes to switching the proto type can we use a boolean flag as a the input to setupRun so like

```golang
proto := protoHTTP
if isTLS {
	proto = protoHTTPS
}
```

This may seem like overkill at the moment but, once you expose the TLS configuration option in the future our users will be calling `s.Run()`. So we'd need to interpret if the user is starting the server with TLS by inspecting the configuration in `Server` so I'd expect to see code like this with `Run()` eventually

```golang
func (s *Server) Run() error {
	s.setupRun(s.TLSConfig != nil)
	return s.Server.ListenAndServe()
}
```

Reply:
OK, this makes sense to me too. But pushing this a step forward if we have a TLSConfig (or a key/cert) we could also have an if inside `Run` for `ListenAndServe` or `ListenAndServeWithTLS`.

I think our initial thoughts around this were to mirror the `net/http` approach but since Fuego's function si called `Run` (no implied connection on TLS or not) I think it will a better developer experience to use options for TLSConfig and cert/key configuration and determine what mode we need to run in based on the options set. I think this would be inline with Fuegos API design (similar pattern to CORS MIddleware but for TLS).  What do you think?

Reply:
> What do you have in mind?

Regarding logging: now that've seen the `disableStartupMessages` option, I think we can leave this as is. If people set the `Addr` up via an environment variable or other config to be something like `localhost:8080` and they have this enabled they can CTRL/CMD+Click the link to open it from a terminal. This might help newcomers and for people that set the listener address to `:8080` but don't have startup messages disabled they can just ignore that.